### ISSUE TITLE
System.Net.NetworkInformation.NetworkChange class throws Exception at Mono.Android

### ISSUE THREAD

**Issue by Saratsin** [mono/mono#8263]
## Steps to Reproduce

1. Create new solution with Mono.Android (most common is Xamarin.Android application)
2. Try to subscribe on System.Net.NetworkInformation.NetworkChange.NetworkAvailabilityChanged event anywhere.
3. Run the code

## Current Behavior
Event subscribing throws an exception: "System.DllNotFoundException: /System/Library/Frameworks/SystemConfiguration.framework/SystemConfiguration"

## Expected Behavior
Event subscribing works as well and firing when network availability has changed

### On which platforms did you notice this

- [x] Android
- [ ] macOS
- [ ] Linux
- [ ] Windows

**Version Used**:
Mono 5.8.1.0

### Stacktrace

```
  at (wrapper managed-to-native) System.Net.NetworkInformation.MacNetworkChange.SCNetworkReachabilityCreateWithAddress(intptr,System.Net.NetworkInformation.MacNetworkChange/sockaddr_in&)
  at System.Net.NetworkInformation.MacNetworkChange..ctor () [0x0000c] in <bcdc1df2b3724ab69797f3819a126346>:0 
  at System.Net.NetworkInformation.NetworkChange.MaybeCreate () [0x00009] in <bcdc1df2b3724ab69797f3819a126346>:0 
```

I think the problem is that somewhy it creating MacNetworkChange instance instead of LinuxNetworkChange on Android. I'm not pretty sure if Linux implementation will also work on Android, just have no idea how to test it.

**Comment by akoeplinger** [comment]
Hmm, this is weird, it should catch any exceptions thrown there and fallback to LinuxNetworkChange: https://github.com/mono/mono/blob/cb08c5d1ac1b77e4c3b5d989cc1e268439913eee/mcs/class/System/System.Net.NetworkInformation/NetworkChange.cs#L110-L116

Which Xamarin.Android version are you using?

**Comment by Saratsin** [comment]
Okay, I've tried to create the sample project, for this bug and found what I mistake I've made and why i had this 'bug'. I was just subscribing `System.Exception` handling in all application, so this when I called this subscription, my code was showing me this exception because firstly mono trying to create the MacNetworkChange class (even though execution would go next after this, because this try/catch block would work). Sorry for this inconvenience. This issue can be closed
